### PR TITLE
Add test cases for JSON resource marshaling

### PR DESCRIPTION
**Pull request by sagar23sj**
Resources Covered :
* Artifact
* ArtifactList

Helps with issue : #55

### CODE REVIEW

**Review by gmlewis — APPROVED** [review]
Thank you, @sagar23sj !
LGTM.

Awaiting second LGTM before merging.

**Comment by gmlewis** [comment]
Friendly ping @wesleimp

**Review by Parker77 — APPROVED** [review]
LGTM.

**Pull request merged**
